### ISSUE TITLE
Fine-tuning setup for full-shot results

### ISSUE THREAD

**Issue by MogicianXD** [Time-MoE/Time-MoE#65]
Dear author,

May I inquire about your setup when fine-tuning the model on six benchmarks. Is the setup consistent with that of zero-shot forecasting, where the context length is 512/1024/2048/3072 for predicting 96/192/336/720 respectively? 

By the way, I find that the results (at least on ETTh datasets) are quite unstable with flash attention during fine-tuning and could be worse than those reported in the paper. I wonder whether your full-shot experiments are conducted under flash attention.

**Comment by Maple728** [comment]
Thank you for your interest in our work!

The default setting of `max_length` in `main.py` is `1024`, which is significantly less than `3072 + 720`. To reproduce the results from the paper, please set `max_length` to `4096`.

PS: To achieve optimal forecasting performance, it is recommended that the sum of context_length and prediction_length does not exceed `max_length`. If you wish to support longer sequence length, please fine-tune Time-MoE with the desired longer sequence length.

Please feel free to contact us, if you have any questions.

**Comment by MogicianXD** [comment]
Sorry, but I'm still not sure how many finetuned models are used for predicting 96/192/336/720. 
Do you mean that we obtain only one model fine-tuned with `max_length=4096` and evaluate it on different horizons? 
If so, which context length is set for predicting 96?  
When following the setting of `512-pred-96`, the performance is not improved. Could you provide your scripts for reproduction?

For details, my script on ETTh2 is
```bash
python torch_dist_run.py main.py -d ETT-small/data_etth2_train_origin.jsonl --precision bf16 --learning_rate 1e-6 --max_length 4096 --attn_implementation flash_attention_2

python run_eval.py -d dataset/ETT-small/ETTh2.csv -p 96 -m logs/time_moe -c 512
python run_eval.py -d dataset/ETT-small/ETTh2.csv -p 96 -m logs/time_moe -c 1024
```
The results are
|        | MSE   | MAE   |
|--------|-------|-------|
| c=512  | 0.307 | 0.361 |
| c=1024 | 0.273 | 0.334 |
| c=2048 | 0.266 | 0.330 |
| c=3072 | 0.267 | 0.330 |

**Comment by MogicianXD** [comment]
When following the setting of `3072-pred-720` and using flash attention, I still fail to reproduce results.
Below are my results under different learning rates:
|   Time-MoE-base  | MSE   | MAE   |
|--------|-------|-------|
| lr=5e-5  | 0.444 | 0.469 |
| lr=1e-5  | 0.420 | 0.455 |
| lr=5e-6 | 0.419 | 0.454 |
| lr=1e-6 | 0.419 | 0.454 |
| lr=1e-7 | 0.419 | 0.454 |
| from paper | 0.404 | 0.431 |

If we finetune the model with `--attn_implementatio eager`, the results are
|   Time-MoE-base  | MSE   | MAE   |
|--------|-------|-------|
| lr=2e-4  | 0.911 | 0.718 |
| lr=1e-4  | 0.402 | 0.448 |
| lr=5e-5  | 0.407 | 0.449 |

**Comment by MogicianXD** [comment]
Updated: The metrics of `3072-pred-720` on all datasets are listed below:

|   Time-MoE-base       | (reproduced)|  | (reported)  | |
|---------|---------------|--------------|---------------|------------|
|         | MSE           | MAE          | MSE           | MAE        |
| ETTh1   | 0.437         | 0.469        | 0.410         | 0.443      |
| ETTh2   | 0.419         | 0.454        | 0.404         | 0.431      |
| ETTm1   | 0.468         | 0.469        | 0.433         | 0.445      |
| ETTm2   | 0.487         | 0.465        | 0.403         | 0.424      |
| weather | 0.377         | 0.400        | 0.352         | 0.366      |

Could you provide more information about how to reproduce full-shot results?

**Comment by Maple728** [comment]
Sorry for the late reply. I have been quite busy recently.

The training datasets used for fine-tuning the model have already been normalized using zero-normalization. Therefore, please set the `normalization` argument to `none` if you use them. In our paper, we fine-tuned Time-MoE for one epoch on the training split of all benchmark datasets. 

This matches our paper's procedure, where we fine-tuned Time-MoE for one epoch on the benchmark training splits. You can reproduce our full-shot results with the following command (I've recently re-run it and achieved similar results):

```bash
python torch_dist_run.py main.py -d {path_to_all_train_datasets} --precision bf16 --learning_rate 1e-4 --min_learning_rate 5e-5 --lr_scheduler_type cosine --attn_implementation flash_attention_2 --model TimeMoE-50M --global_batch_size 128 --normalization none
```

I will update the default hyperparameters in `main.py` to help the community fine-tune a better Time-MoE model for their own forecasting scenarios.

**Comment by MogicianXD** [comment]
Thanks for your reply! Wish everything goes well with your work！

Do you mean that we finetune the model on the combination of ETTh1, ETTh2, ETTm1, ETTm2, etc.. and run the test on a single benchmark (e.g., ETTh1)? Would it be fair for the baselines (e.g., PatchTST) that are trained on only one dataset.

**Comment by Maple728** [comment]
> Thanks for your reply! Wish everything goes well with your work！
> 
> Do you mean that we finetune the model on the combination of ETTh1, ETTh2, ETTm1, ETTm2, etc.. and run the test on a single benchmark (e.g., ETTh1)? Would it be fair for the baselines (e.g., PatchTST) that are trained on only one dataset.

In other time-series models, the standard practice is to train and evaluate using a single dataset per benchmark, which often leads to better performance on each specific benchmark.

In contrast, Time-MoE is trained on all available datasets and subsequently tested on each benchmark. Generally, this approach tends to have certain disadvantages compared to the former.

**Comment by MogicianXD** [comment]
Thanks for your answer!
